components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,199 @@
+
+    private EntityReferenceResolver<EntityReference> currentResolver;
+
+    private DomainObjectFactory factory;
+
+    @Mock
+    private Patient patient;
+
+    private URI uri;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private DefaultPatientsResourceImpl patientsResource;
+
+    private XWikiContext context;
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager compManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(compManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.patientsResource = (DefaultPatientsResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.queries = this.mocker.getInstance(QueryManager.class);
+        this.uri = new URI("http://uri");
+
+        doReturn(this.uri).when(this.uriInfo).getBaseUri();
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+        ReflectionUtils.setFieldValue(this.patientsResource, "uriInfo", this.uriInfo);
+
+        doReturn("00000001").when(this.patient).getId();
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
+    }
+
+    @Test
+    public void addPatientUserDoesNotHaveAccess() throws XWikiRestException {
+        WebApplicationException exception = new WebApplicationException();
+        doReturn(false).when(this.access).hasAccess(Right.EDIT, null, mock(EntityReference.class));
+        try {
+            Response response = this.patientsResource.addPatient("");
+        }
+        catch (WebApplicationException ex){
+            exception = ex;
+        }
+        Assert.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), exception.getResponse().getStatus());
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", "");
+    }
+
+    @Test
+    public void addNullPatient() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        Response response = this.patientsResource.addPatient(null);
+        Assert.assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+        verify(this.logger).error(eq("Could not process remote matching request: {}"), anyString(), anyObject());
+    }
```

Comment:
Again, the Exception handling behaviour you are testing for here is occurring, but only because of a NullPointerException generated after `this.repository.createNewPatient();`. You need to mock the behaviour of this method call. Note that if you pass a null value to `addPatient()` the line `Patient patient = this.repository.createNewPatient();` will still execute since `JSONObject.fromObject(json);` will return a new JSONObject when passed a null value